ksqldb-engine/src/main/java/io/confluent/ksql/engine/EngineExecutor.java

```diff
@@ -762,7 +851,59 @@ private ExecutorPlans planQuery(
           queryId,
           oldPlanInfo
       );
-      return new ExecutorPlans(logicalPlan, physicalPlan);
+
+      return new ExecutorPlans(logicalPlan.getNode().get(), physicalPlan);
+    }
+  }
+
+  private static class StubbedOutputNode extends KsqlStructuredDataOutputNode {
```

Comment:
Still might need further refactoring. The problem is, that we need to pass some "logical plan" (old format) into `ExecutorPlans` (so we just use a mock for now, as we don't really generate the old logical plan that we can pass in).

Not sure to what extend we can modify `ExecutorPlans` later to avoid this -- it's really just to make the code work for now; not a permanent solution.